plugins/modules/aci_interface_policy_storm_control.py

```diff
@@ -0,0 +1,433 @@
+    ACIModule,
+    aci_argument_spec,
+    aci_annotation_spec,
+    aci_owner_spec,
+    storm_control_policy_rate_spec,
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        storm_control_policy=dict(
+            type="str", required=False, aliases=["name", "storm_control", "storm_control_name"]
+        ),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+        all_types_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["all_types"]),
+        broadcast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["broadcast"]),
+        multicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["multicast"]),
+        unicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["unicast"]),
+        storm_control_action=dict(type="str", choices=["drop", "shutdown"]),
+        storm_control_soak_action=dict(type="int"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["storm_control_policy"]],
+            ["state", "present", ["storm_control_policy"]],
+        ],
+        mutually_exclusive=[
+            ("all_types_configuration", "broadcast_configuration"),
+            ("all_types_configuration", "multicast_configuration"),
+            ("all_types_configuration", "unicast_configuration"),
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    storm_control_policy = module.params.get("storm_control_policy")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    storm_control_action = module.params.get("storm_control_action")
+    storm_control_soak_action = module.params.get("storm_control_soak_action")
+    all_types_configuration = module.params.get("all_types_configuration")
+    broadcast_configuration = module.params.get("broadcast_configuration")
+    multicast_configuration = module.params.get("multicast_configuration")
+    unicast_configuration = module.params.get("unicast_configuration")
+
+    rates_input = {}
+
+    def get_rates_configuration(configuration, percentage, pps, burst_percentage, burst_pps):
+        if configuration is None:
+            return {}
+        rate = configuration.get("rate")
+        burst_rate = configuration.get("burst_rate")
+        rate_type = configuration.get("rate_type")
+
+        if rate_type == "percentage":
+            for rate_name, rate_value in dict(rate=rate, burst_rate=burst_rate).items():
```

Comment:
Why do we create this for loop? Not sure I understand why not use rate and burst rate directly in your condition and error handling.